spark/src/test/scala/org/apache/spark/sql/delta/coordinatedcommits/UCCommitCoordinatorClientSuite.scala

```diff
@@ -0,0 +1,362 @@
+
+  @Mock
+  private val mockFactory: UCClientFactory = mock(classOf[UCClientFactory])
+
+  private var ucClient: UCClient = _
+
+  private var ucCommitCoordinator: InMemoryUCCommitCoordinator = _
+
+  protected override def sparkConf = super.sparkConf
+      .set("spark.sql.catalog.main", "io.unitycatalog.connectors.spark.UCSingleCatalog")
+      .set("spark.sql.catalog.main.uri", "https://test-uri.com")
+      .set("spark.sql.catalog.main.token", "test-token")
+      .set("spark.hadoop.fs.file.impl", classOf[LocalFileSystem].getCanonicalName)
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    tableUUID = UUID.randomUUID().toString
+    UCCommitCoordinatorClient.BACKFILL_LISTING_OFFSET = 100
+    metastoreId = UUID.randomUUID().toString
+    DeltaLog.clearCache()
+    Mockito.reset(mockFactory)
+    CommitCoordinatorProvider.clearAllBuilders()
+    UCCommitCoordinatorBuilder.ucClientFactory = mockFactory
+    UCCommitCoordinatorBuilder.clearCache()
+    CommitCoordinatorProvider.registerBuilder(UCCommitCoordinatorBuilder)
+    ucCommitCoordinator = new InMemoryUCCommitCoordinator()
+    ucClient = new InMemoryUCClient(metastoreId, ucCommitCoordinator)
+    when(mockFactory.createUCClient(anyString(), anyString())).thenReturn(ucClient)
+  }
+
+  override protected def commit(
+      version: Long,
+      timestamp: Long,
+      tableCommitCoordinatorClient: TableCommitCoordinatorClient): JCommit = {
+    val commitResult = super.commit(version, timestamp, tableCommitCoordinatorClient)
+    // As backfilling for UC happens after every commit asynchronously, we block here until
+    // the current in-progress backfill has completed in order to make tests deterministic.
+    eventually(timeout(10.seconds)) {
+      val logPath = tableCommitCoordinatorClient.logPath
+      val log = DeltaLog.forTable(spark, JCoordinatedCommitsUtils.getTablePath(logPath))
+      val fs = logPath.getFileSystem(log.newDeltaHadoopConf())
+      assert(fs.exists(FileNames.unsafeDeltaFile(logPath, version)))
+    }
+    commitResult
+  }
+
+  override protected def createTableCommitCoordinatorClient(
+      deltaLog: DeltaLog): TableCommitCoordinatorClient = {
+    var commitCoordinatorClient = UCCommitCoordinatorBuilder
+      .build(spark, Map(UCCommitCoordinatorClient.UC_METASTORE_ID_KEY -> metastoreId))
+      .asInstanceOf[UCCommitCoordinatorClient]
+    commitCoordinatorClient = new UCCommitCoordinatorClient(
+      commitCoordinatorClient.conf,
+      commitCoordinatorClient.ucClient) with DeltaLogging {
+      override def recordDeltaEvent(opType: String, data: Any, path: Path): Unit = {
+        data match {
+          case ref: AnyRef => recordDeltaEvent(null, opType = opType, data = ref, path = Some(path))
+          case _ => super.recordDeltaEvent(opType, data, path)
+        }
+      }
+    }
+    // Initialize table ID for the calling test
+    // tableUUID = UUID.randomUUID().toString
```

Comment:
It looks like we have tableUUID as a class member so we can remove this comment (lines 140 - 141)?